src/Trip/Manipulator.ts

```diff
@@ -76,6 +76,42 @@ export function addDaysToTrip(
 	return resolveStickiness(trip, userSettings);
 }
 
+export function addDayToTrip(
+	trip: Trip,
+	dayIndex: number,
+	userSettings: UserSettings | null
+): Trip {
+	checkDayExists(trip, dayIndex);
+	if (!trip.endsOn) {
+		throw new Error('endsOn property in Trip cannot be null');
+	}
+
+	let resultTrip = cloneDeep(trip);
+	resultTrip.days.splice(dayIndex, 0, {
+		itinerary: [],
+		note: null,
+		date: null
+	} as Day);
+	resultTrip.endsOn = resultTrip.endsOn ? addDaysToDate(resultTrip.endsOn, 1) : resultTrip.endsOn;
+	resultTrip.days = decorateDaysWithDate(resultTrip.startsOn, resultTrip.days);
+
+	if (dayIndex > 0 && resultTrip.days[dayIndex - 1].itinerary.length > 0) {
+		const previousDayLastItem = resultTrip.days[dayIndex - 1].itinerary[resultTrip.days[dayIndex - 1].itinerary.length - 1];
+		if (previousDayLastItem.place && previousDayLastItem.isStickyLastInDay) {
+			resultTrip = addPlaceToDay(resultTrip, previousDayLastItem.place, dayIndex, userSettings);
+		}
+
+	}
+	if (dayIndex !== 0 && resultTrip.days[dayIndex + 1].itinerary.length > 0) {
```

Comment:
you cannot be sure the trip was not empty. 🤔

Reply:
Here I can. If it was empty, it would fail on `checkDayExists()`. If there was 1 day, it would fail on `dayIndex !== 0`. Any more days, even if adding at last position it's okay, because it's inserting (splicing) new day at given position, so there always is at least one day after one that's inserted.